File: 1582. Special Positions in a Binary Matrix.py
class Solution:
    def numSpecial(self, grid: List[List[int]]) -> int:
        row, col = [], []
        n, m = len(grid), len(grid[0])
        for i in range(n):
            row.append(grid[i].count(1))
        for j in range(m):
            count = 0
            for i in range(n):
                count += grid[i][j]
            col.append(count)
        
        ans = 0
        for i in range(n):
            for j in range(m):
                ans += grid[i][j] == 1 and row[i] == 1 and col[j] == 1
        return ans

File: 1582. Special Positions in a Binary Matrix.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) 
    {
        int n = mat.size(), m = mat[0].size(), ans = 0;
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<m;j++)
            {
                if(mat[i][j]==1)
                {
                    int cnt = 0;
                    for(int k=0;k<m;k++)
                        if(mat[i][k]==1) cnt++;
                    for(int k=0;k<n;k++)
                        if(mat[k][j]==1) cnt++;
                    if(cnt==2) ans++;
                }
            }
        }
        return ans;
    }
};